packages/commerce-sdk-react/src/hooks/ShopperBaskets/cache.ts

```diff
@@ -0,0 +1,186 @@
+const getBasePath = (parameters: {organizationId: string}) => [
+    '/organizations/',
+    parameters.organizationId
+]
+const getBasketPath = (parameters: GetBasketParameters) => [
+    ...getBasePath(parameters),
+    '/baskets/',
+    parameters.basketId
+]
+const getCustomerBasketsPath = (parameters: GetCustomerBasketsParameters) => [
+    ...getBasePath(parameters),
+    '/customers/',
+    parameters.customerId,
+    '/baskets' // No trailing / as it's an aggregate endpoint
+]
+
+// Parameters helpers
+/**
+ * Creates an object with *only* the parameters from `getBasket`, omitting unwanted parameters from
+ * other endpoints. (Extra parameters can break query key matching.)
+ */
+const toGetBasketParameters = (parameters: GetBasketParameters): GetBasketParameters =>
+    pick(parameters, ['basketId', 'locale', 'organizationId', 'siteId'])
+/**
+ * Creates an object with *only* the parameters from `getBasket`, omitting unwanted parameters from
+ * other endpoints. (Extra parameters can break query key matching.)
+ */
+const toGetCustomerBasketsParameters = (
+    customerId: string,
+    parameters: Omit<GetCustomerBasketsParameters, 'customerId'>
+): GetCustomerBasketsParameters =>
+    pick({customerId, ...parameters}, ['customerId', 'organizationId', 'siteId'])
+
+const updateBasketQuery = (
+    customerId: string | null,
+    parameters: BasketParameters,
+    newBasket: Basket
+): CacheUpdate => {
+    // The `parameters` received includes the client config and parameters from other endpoints
+    // so we need to exclude unwanted parameters in the query key
+    const getBasketParameters = toGetBasketParameters(parameters)
+    const basketUpdate = {
+        queryKey: [...getBasketPath(parameters), getBasketParameters] as const
+    }
+
+    // We can only update customer baskets if we have a customer!
+    if (!customerId) return {update: [basketUpdate]}
+
+    // Similar elision as done for `getBasket`
+    const getCustomerBasketsParameters = toGetCustomerBasketsParameters(customerId, parameters)
+    const customerBasketsUpdate = {
+        queryKey: [
+            ...getCustomerBasketsPath(getCustomerBasketsParameters),
+            getCustomerBasketsParameters
+        ] as const,
+        updater: (oldData?: CustomerBasketsResult): CustomerBasketsResult | undefined => {
+            // do not update if response basket is not part of existing customer baskets
+            if (!oldData?.baskets?.some((basket) => basket.basketId === parameters.basketId)) {
+                return undefined
+            }
+            const updatedBaskets = oldData.baskets.map((basket) => {
+                return basket.basketId === parameters.basketId ? newBasket : basket
+            })
+            return {
+                ...oldData,
+                // Shopper Customers and Shopper Baskets have different definitions for the `Basket`
+                // type. (99% similar, but that's not good enough for TypeScript.)
```

Comment:
😢

Reply:
If yah'll can describe this problem with me directly, we can consider raising it with the relevant teams to align the type definitions.

Reply:
We discussed in slack last week, I think it was. I raised the issue with Pratik, but said it's not urgent. You can follow up and raise the priority, though.